internal/pproflogging/pproflogging_test.go

```diff
@@ -124,3 +219,140 @@ func TestDebug_newProfileConfigs(t *testing.T) {
 		})
 	}
 }
+
+func TestDebug_LoadProfileConfigs(t *testing.T) {
+	// save environment and restore after testing
+	saveLockEnv(t)
+
+	ctx := context.Background()
+
+	tcs := []struct {
+		inArgs                       string
+		profileKey                   ProfileName
+		profileFlagKey               string
+		expectProfileFlagValue       string
+		expectProfileFlagExists      bool
+		expectConfigurationCount     int
+		expectError                  error
+		expectProfileConfigNotExists bool
+	}{
+		{
+			inArgs:                       "",
+			expectConfigurationCount:     0,
+			profileKey:                   "",
+			expectError:                  nil,
+			expectProfileConfigNotExists: true,
+		},
+		{
+			inArgs:                   "block=rate=10:cpu:mutex=10",
+			expectConfigurationCount: 3,
+			profileKey:               "block",
+			profileFlagKey:           "rate",
+			expectProfileFlagExists:  true,
+			expectProfileFlagValue:   "10",
+			expectError:              nil,
+		},
+		{
+			inArgs:                   "block=rate=10:cpu:mutex=10",
+			expectConfigurationCount: 3,
+			profileKey:               "cpu",
+			profileFlagKey:           "rate",
+			expectProfileFlagExists:  false,
+		},
+		{
+			inArgs:                   "block=rate=10:cpu:mutex=10",
+			expectConfigurationCount: 3,
+			profileKey:               "mutex",
+			profileFlagKey:           "10",
+			expectProfileFlagExists:  true,
+		},
+		{
+			inArgs:                       "mutex=10",
+			expectConfigurationCount:     1,
+			profileKey:                   "cpu",
+			profileFlagKey:               "10",
+			expectProfileConfigNotExists: true,
+		},
+	}
+
+	for i, tc := range tcs {
+		t.Run(fmt.Sprintf("%d: %q", i, tc.inArgs), func(t *testing.T) {
+			pmp, err := LoadProfileConfig(ctx, tc.inArgs)
+			require.ErrorIs(t, tc.expectError, err)
+			if err != nil {
+				return
+			}
+			val, ok := pmp[tc.profileKey]
+			require.Equalf(t, tc.expectProfileConfigNotExists, !ok, "expecting key %q to %t exist", tc.profileKey, !tc.expectProfileConfigNotExists)
+			if tc.expectProfileConfigNotExists {
+				return
+			}
+			flagValue, ok := val.GetValue(tc.profileFlagKey)
+			require.Equal(t, tc.expectProfileFlagExists, ok, "expecting key %q to %t exist", tc.profileKey, tc.expectProfileFlagExists)
+			if tc.expectProfileFlagExists {
+				return
+			}
+			require.Equal(t, tc.expectProfileFlagValue, flagValue)
+		})
+	}
+}
+
+//nolint:gocritic
+func saveLockEnv(t *testing.T) {
+	t.Helper()
+
+	oldEnv := os.Getenv(EnvVarKopiaDebugPprof)
+
+	t.Cleanup(func() {
+		// restore the old environment
+		t.Setenv(EnvVarKopiaDebugPprof, oldEnv)
```

Comment:
It seems that this has little or no effect.
`t.Setenv()` will internally setup a cleanup function to reset the value to whatever value it found when `t.Setenv()` was called.

In this case, `t.Setenv()` is being called as part of the cleanup, so this is after the test has executed. Then
- it will read the current value for the env var (`v1`), which may have been set during the test
- set the env var value to `oldEnv`
- setup a `t.Cleanup` function to reset the env var back to `v1`. It may be the case that this newly set cleanup function does not execute at all. I do not know what is the expected behavior or calling `t.Cleanup()` from within a `t.Cleanup()` callback.

https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/testing/testing.go;l=1299

The suggestion is to use `t.Setenv` directly in the tests, instead of the `saveLockEnv` implementation. 

Related, TIL, `t.Setenv` ensures that neither the test, nor any of is ancestors (in the case of a subtest) are executed in parallel, which is what is desired in this case.